### ISSUE TITLE
IOUtils.close throws NPE

### ISSUE THREAD

**Issue by TobiasKiecker** [Atmosphere/atmosphere#2515]
**Describe the bug**
The `IOUtils.close` function: 
https://github.com/Atmosphere/atmosphere/blob/d350bc5a8a46663fd54d82ba545dcfa5854f66b0/modules/cpr/src/main/java/org/atmosphere/util/IOUtils.java#L463-L486 
throws a null pointer exception if it is called with one single parameter which is null.

**Atmosphere Info**
  - version: all since b31804c4a8fc393f953464fb23c0de7fd8c36323
  - atmosphere.js version NA
  - extensions used None

**Expected behavior**
Should not throw null.

**Screenshots**

**Systems (please complete the following information):**
 - OS: NA
 - Browser name and version: NA
 - Java version and distribution: at least Java 21 probably more
 - Serveur name and version: NA

**Additional context**
This is an edge case of Java behavior of varargs.

An improved implementation could be:

```
if (closeableArray != null) {
    for (Closeable closeable : closeableArray) {
        if (closeable != null) {
            try {
                closeable.close();
            } catch (IOException ioe) {
                logger.info("Can't close the object", ioe);
            }
        }
    }
}
```

**Comment by slovdahl** [comment]
Good catch. Just out of curiosity, have you seen this cause problems in reality?

**Comment by TobiasKiecker** [comment]
No not really, we found it as a part of a research project.
